### PR TITLE
Add supoort for enabling GPU access

### LINKED ISSUE #919: FR: Enabling GPU access to service containers
Currently, the following configuration is not work on podman-compose.

```yaml
services:
  test:
    image: nvidia/cuda:12.3.1-base-ubuntu20.04
    command: nvidia-smi
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
```

https://docs.docker.com/compose/gpu-support/

### PR DESCRIPTION
**Pull request by mokeyish**
closes #919

### CODE REVIEW

**Review by p12tic — CHANGES_REQUESTED** [review]
Looks good, thanks.

Could you add unit tests? There's a bunch already in pytests/test_container_to_args.py. It should be easy to extend the test suite for the GPU options. The existing tests test `container_to_args()` which I think is appropriate level of detail for the additional GPU tests.

**Comment by mokeyish** [comment]
@p12tic  I looked at the existing unit tests and it was a bit difficult to understand. It seems to be output after execution. But Github Actions does not have a GPU and cannot be tested.

**Comment by p12tic** [comment]
@mokeyish Tests in pytests/test_container_to_args.py don't actually run any containers.

The test structure is as follows:
 - create compose definition in parsed form (i.e. python dicts, lists, etc. are used)
 - run it through container_to_args function
 - the result is arguments which would be otherwise sent to podman. The test simply compares them to what's expected and does not run podman.
 
With such tests any feature of podman-compose can be tested without any requirements from hardware.

**Comment by p12tic** [comment]
So in GPU tests I would expect tests that check whether correct GPU-related options are added to podman_args.

File: pytests/test_container_to_args.py [hunk 1]
```diff
@@ -325,3 +325,106 @@ async def test_env_file_obj_optional(self):
                 "busybox",
             ],
         )
+
+    async def test_gpu(self):
```
Comment (p12tic) [inline pytests/test_container_to_args.py:329]: This test is actually 4 tests. Could you split it. It's ok if the preamble

```
        c = create_compose_mock()

        cnt = get_minimal_container()
        cnt["command"] = ["nvidia-smi"]
        cnt["deploy"] = {"resources": {"reservations": {"devices": [{}]}}}
```

repeats in each test.

**Pull request merged**